yt_dlp/extractor/zdf.py

```diff
@@ -410,32 +413,72 @@ def _og_search_title(self, webpage, fatal=False):
         title = super()._og_search_title(webpage, fatal=fatal)
         return re.split(r'\s+[-|]\s+ZDF(?:mediathek)?$', title or '')[0] or None
 
+    def _extract_document_id(self, webpage):
+        matches = re.search(r'docId\s*:\s*[\'"](?P<docid>[^\'"]+)[\'"]', webpage)
+        return matches and matches.group('docid')
+
+    def _get_playlist_description(self, page_data):
+        headline = traverse_obj(page_data, ('shortText', 'headline'))
+        text = traverse_obj(page_data, ('shortText', 'text'))
+        if headline is not None and text is not None:
+            return f'{headline}\n\n{text}'
+        return headline or text
+
+    def _convert_thumbnails(self, thumbnails):
+        return [{
+            'id': key,
+            'url': thumbnail_info['url'],
+            'width': int_or_none(thumbnail_info.get('width')),
+            'height': int_or_none(thumbnail_info.get('height')),
+        } for key, thumbnail_info in thumbnails.items() if url_or_none(thumbnail_info.get('url'))]
+
+    def _teaser_to_url_result(self, teaser):
+        return self.url_result(
+            teaser['sharingUrl'], ie=ZDFIE.ie_key(),
+            id=teaser.get('id'), title=teaser.get('titel', ''),
+            thumbnails=self._convert_thumbnails(teaser.get('teaserBild', {})),
```

Comment:
if there's only 1 thumbnail IMO it can just be `thumbnail=thumb_url`

Reply:
There is always a bunch of them at various resolutions as far as I could see